### ISSUE TITLE
Evaluating a complex function silently gives wrong result

### ISSUE THREAD

**Issue by dpsanders** [JuliaApproximation/ApproxFun.jl#700]
I probably shouldn't be doing this like this but I'm trying to create a polynomial with complex coefficients:

```jl
julia> x = Fun(identity, (-100)..100)
Fun(Chebyshev(-100..100),[0.0, 100.0])

julia> f = im*x + 2
Fun(Chebyshev(-100..100),Complex{Float64}[2.0 + 0.0im, 0.0 + 100.0im])

julia> f(1)   # correct
2.0 + 1.0im    

julia> f(im)   # wrong 
0.0 + 0.0im
```

**Comment by dlfivefifty** [comment]
Use ` extrapolate(f,im) ` to evaluate off the domain of definition. Note for high order polynomials this has accuracy issues

**Comment by dpsanders** [comment]
So then shouldn't an error be thrown when you evaluate at a point not in the domain, instead of silently doing the wrong thing? That would be nice.

**Comment by dpsanders** [comment]
And BTW what's the correct way to define a polynomial over a complex domain? Thanks!

**Comment by dlfivefifty** [comment]
> So then shouldn't an error be thrown when you evaluate at a point not in the domain, instead of silently doing the wrong thing?

Yes there should be an error, it was a poor decision made a long time ago that I haven’t corrected

**Comment by dlfivefifty** [comment]
> And BTW what's the correct way to define a polynomial over a complex domain

What kind of domain? On a circle you can do ` Fun(f, Taylor(Circle(z0,r))) `

**Comment by dpsanders** [comment]
Thanks. But maybe some missing conversions?:

```jl
julia> z0 = 0 + 0im
0 + 0im

julia> r = 10
10

julia> x = Fun(f, Taylor(Circle(z0,r)))
ERROR: MethodError: no method matching Int64(::Irrational{:π})
```

It works when I make everything a Float64.
